### PR TITLE
Voting system

### PR DESCRIPTION
**Pull request by robertsapunarich**
I don't think this is quite ready to merge, especially because the voting functionality isn't in the views yet, but I wanted to make this available for code review. I tried to be clever and build my own voting system from scratch, but settled on using the `acts_as_votable` gem. The documentation can be found [here](https://github.com/ryanto/acts_as_votable). I'm also referring to [this tutorial](https://masteruby.github.io/weekly-rails/2014/08/05/how-to-add-voting-to-rails-app.html#.VrKPODYrJ0d) quite a bit.

### CODE REVIEW

**Comment by kenmazaika** [comment]
Looks solid to me, @robertsapunarich

**Comment by robertsapunarich** [comment]
I think this is ready to go now. Wrote a method in the model that seemed to solve everything. If anyone has suggestions for refactoring or wants to evaluate the code from an OO design perspective, I'm open.

File: app/models/user.rb [hunk 1]
```diff
@@ -26,4 +27,9 @@ def self.search(term)
     # could use ILIKE since we're using postgres to drop downcase
     where('LOWER(email) LIKE :term OR LOWER(name) LIKE :term', term: "%#{term.downcase}%")
   end
+
+  # Check if a user voted on a post. Cf. /views/posts/index.html.haml line 63.
```
Comment (kenmazaika) [inline app/models/user.rb:31]: Smart move pushing a helper method in the model.  :+1:

File: app/views/posts/index.html.haml [hunk 2]
```diff
@@ -53,7 +53,14 @@
...
         =link_to image_tag(post.user.gravatar_url(default: "retro", size: 50), alt: post.user.name, class: "gravatar pull-left"), user_path(post.user)
-      .col-sm-10.col-xs-12
+      .col-sm-9.col-xs-12
         %p= post.created_at.to_formatted_s(:long)
         - if post.user.present?
           %small= post.user.email
+      .col-sm-1.col-xs-12
+        %p= "#{post.votes_for.size} ranks of awesomesauce!"
+      .col-sm-1.col-xs-12
+        - if current_user.upvoted? post
```
Comment (kenmazaika) [inline app/views/posts/index.html.haml:63]: I think the `?` vs. `!` is backwards?

**Comment by kenmazaika** [comment]
Thumbs up @robertsapunarich.  Awesome code!

**Pull request merged**